src/exaerror.lua

```diff
@@ -0,0 +1,164 @@
+local M = {
+    VERSION = "1.0.0",
+}
+
+-- Lua 5.1 backward compatibility
+_G.unpack = table.unpack or _G.unpack
+
+---
+-- Convert error to a string representation.
+-- <p>
+-- Note that <code>__tostring</code> is the metamethod called by Lua's global <code>tostring</code> function.
+-- This allows using the error message in places where Lua expects a string.
+-- </p>
+--
+-- @return string representation of the error object
+--
+function M:__tostring()
+    local pieces = {}
+    if self.message then
+        table.insert(pieces, self:get_message())
+    end
+    if self.mitigations then
+        for _, mitigation in ipairs(self.mitigations) do
+            table.insert(pieces, mitigation)
+        end
+    end
+    if self.code then
+        if #pieces > 0 then
+            table.insert(pieces, 1, self.code .. ":")
+        else
+            table.insert(pieces, 1, self.code)
+        end
+    end
+    if #pieces > 0 then
+        return table.concat(pieces, " ")
+    else
+        return "Undefined error. This should not happen. Please report to the software maker."
+    end
+end
+
+---
+-- Concatenate an error object with another object.
+--
+-- @return String representing the concatenation.
+--
+function M.__concat(left, right)
+    return tostring(left) .. tostring(right)
```

Comment:
Why do you need this?

Reply:
Because otherwise you get an error that you tried to append a raw table (the error object). This way it works just fine.